vita3k/shader/src/translator/ialu.cpp

```diff
@@ -121,10 +115,27 @@ bool USSETranslatorVisitor::vbw(
     case Opcode::SHR: operation = spv::Op::OpShiftRightLogical; break;
     default: return false;
     }
-
-    result = m_b.createBinOp(operation, type_ui32, src1, src2);
-    if (m_b.isFloatType(m_b.getTypeId(src2))) {
-        result = m_b.createUnaryOp(spv::Op::OpBitcast, type_f32, src2);
+    //optimisation. (any OR 0 || any XOR 0 || any AND 0xFFFFFFFF) -> assign
+    bool is_const = m_b.getOpCode(src2) == spv::Op::OpConstant;
+    if ((operation == spv::Op::OpBitwiseOr || operation == spv::Op::OpBitwiseXor) && is_const && (m_b.getConstantScalar(src2) == 0)
```

Comment:
if (((operation == spv::Op::OpBitwiseOr || operation == spv::Op::OpBitwiseXor) && is_const && (m_b.getConstantScalar(src2) == 0))
        || ((operation == spv::Op::OpBitwiseAnd) && is_const && (m_b.getConstantScalar(src2) == 0xFFFFFFFF))) {

You must add more brackets if you want compiler to recognise it properly I think. If you are following mr zang suggestion.